tests/test_restrictions_emmo.py

```diff
@@ -15,10 +23,137 @@
 class TestRestrictionsEmmo(unittest.TestCase):
     """Test the restrictions in EMMO ontology."""
 
-    def test_emmo_datatypes(self):
-        """Test for classes Integer, Boolean, .. in EMMO."""
-        for r in math.Integer.axioms:
-            print(r)  # TODO finish tests
+    emmo_datatypes = (math.Integer, math.Real, math.Boolean, math.Vector,
+                      math.Matrix)
+
+    quantifiers = {rdflib.OWL.someValuesFrom: QUANTIFIER.SOME,
+                   rdflib.OWL.allValuesFrom: QUANTIFIER.ONLY,
+                   rdflib.OWL.cardinality: QUANTIFIER.EXACTLY,
+                   rdflib.OWL.minCardinality: QUANTIFIER.MIN,
+                   rdflib.OWL.maxCardinality: QUANTIFIER.MAX}
+
+    rtypes = {OntologyRelationship: RTYPE.RELATIONSHIP_RESTRICTION,
+              OntologyAttribute: RTYPE.ATTRIBUTE_RESTRICTION}
+
+    def iter_restrictions(self):
+        """Returns all the datatypes of the test wrapped in an iterator."""
+        restrictions = iter([])
+        for datatype in self.emmo_datatypes:
+            restrictions = itertools.chain(restrictions,
+                                           (r for r in datatype.axioms
+                                            if type(r) is Restriction))
+        return restrictions
+
+    def test___str__(self):
+        """Test the __str__ method."""
+        for r in self.iter_restrictions():
+            self.assertIs(type(r.__str__()), str)
+
+    def test_quantifier(self):
+        """Test the quantifier method.
+
+        Also tests _compute_target and _check_quantifier implicitly.
+        """
+        for r in self.iter_restrictions():
+            b_node = r._bnode
+            graph = r._graph
+
+            restriction_predicates = graph.predicates(subject=b_node)
+            quantifiers = [self.quantifiers[p] for p in restriction_predicates
+                           if p in self.quantifiers]
+            self.assertTrue(len(quantifiers) == 1, "The restriction has not "
```

Comment:
Hi @kysrpex 
in unittest2, you can use 
self.assertEqual(arg1, arg2) to check if arg1 equals arg2.
I think the benefit of using assertEqual compared to assertTrue is that the former prints more info when the two are not the same

Reply:
Thank you, changed.